src/ts/main.ts

```diff
@@ -0,0 +1,215 @@
+import $ from "jquery";
+import {invoke} from "@tauri-apps/api/core";
+import {open as openDialog} from "@tauri-apps/plugin-dialog";
+import {open as openShell} from "@tauri-apps/plugin-shell";
+import {listen} from "@tauri-apps/api/event";
+
+function setLoader(state: boolean) {
+	$("#busy").prop("hidden", !state);
+}
+
+
+function setLoadingState(state: boolean) {
+	$("#busy").prop("hidden", !state);
+
+	// loop through all buttons and input fields and disable them
+	for (const element of document.querySelectorAll("button, input")) {
+		if (element.closest("#settings-content")) continue;
+		(element as any).disabled = state;
+	}
+
+	// These elements need additional properties to be properly disabled
+	$("#pickpath").prop("ariaDisabled", state);
+	$("#downloadbtn").prop("ariaDisabled", state);
+
+	// disable internet buttons
+	for (const element of document.querySelectorAll("#internet-btns div")) {
+		element.ariaDisabled = String(state);
+	}
+}
+
+
+/// Returns list of IDs of invalid form fields
+const invalidFields = () => {
+	const form = document.forms[0];
+
+	const invalidFields: string[] = [];
+	for (const input of form) {
+		const inputElement = input as HTMLInputElement;
+		const valid = !(inputElement.value === "" && inputElement?.parentElement?.classList.contains("required"));
+		if (!valid) {
+			invalidFields.push(inputElement.id);
+		}
+	}
+	// console.debug(`[${invalidFields.join(", ")}] fields invalid/empty`);
+
+	return invalidFields;
+};
+
+
+$(async () => {
```

Comment:
You are already checking for all available terminals and disableling. could modifi this together with the rust to read from some kind of file with all the terminals and generate it. 
Looks like something that would be forgotton easily when adding/removing options.